scripts/verify-agent-version-update.sh

```diff
@@ -9,7 +9,7 @@ fi
 # Find the common ancestor between the current set of changes and the upstream branch, then see if any source code files
 # have changed in the agent or command packages.
 common_ancestor=$(git merge-base ${BRANCH_NAME}@{upstream} HEAD);
-if [[ "$(git diff --name-only ${common_ancestor} -- agent/**.go agent/**/*.go ':!agent/**_test.go' ':!agent/**/*_test.go' command/**.go command/**/*.go ':!command/**_test.go' ':!command/**/*_test.go')" == "" ]]; then
+if [[ "$(git diff --name-only ${common_ancestor} -- agent/**.go agent/**/*.go ':!agent/**_test.go' ':!agent/**/*_test.go' command/**.go command/**/*.go ':!command/**_test.go' ':!command/**/*_test.go') rest/client/**.go ':!rest/client/**_test.go' ':!rest/client/client.go'" == "" ]]; then
```

Comment:
Why is the rest client required for the agent update? This seems like it would produce a lot of false positives because the communicator can change if the communicator is changed for the CLI. The intention of this test was not to be exhaustive in finding all possible changes to the agent.

Reply:
my intent was to just have this match the parts of the rest client that were created for the agent, but the check was too inclusive. I've changed the approach to just list the files made for the agent

Reply:
I also hesitate to do that because the CLI/agent files overlap and it seems fragile to cherry-pick individual files in  our repo for the agent update test. Since the agent update tests doesn't block merging, I feel like people would start ignoring the agent update test if it gave them false positives too often (which this could do) and I intentionally only picked files that definitely would affect the agent - otherwise, you could argue that it should check every single DB model file as well.

Reply:
Right, the ones I added are used exclusively by the agent. I also removed requests.go just now because that's shared. But I think the ones here are logically part of the agent, but just in a different package due to dependencies

Reply:
I think this organization problem will go away this sprint so we won't need this I made [EVG-13233](https://jira.mongodb.org/browse/EVG-13233) with the intention of moving the agent REST communicator code out of the client package (and maybe as a subpackage of the agent).